app/controllers/api/v1/stats/channels_controller.rb

```diff
@@ -0,0 +1,65 @@
+class Api::V1::Stats::ChannelsController < Api::BaseController
+
+  def channels
+
+    if(params[:uuid] == nil)
+
+      channels = Channel.all.map { |channel| channel.id }
+      data = JSON.pretty_generate(channels)
+      render(json: data)
+
+    else
+
+      channel = Channel.find_by_id(params[:uuid])
+
+      if(channel == nil)
```

Comment:
This is personal preference so it's up to you, but i'd rather have the `.find` raise an exception and then we catch it in this method instead of checking for nil and having code be more nested.

Reply:
Agree, was debating this before, since it does add some more code, but in the future we'll probably want to check for more errors so it's good to lay a foundation for handling, Fixed.